Mathlib/Analysis/LocallyConvex/WeakSpace.lean

```diff
@@ -0,0 +1,101 @@
+/-
+Copyright (c) 2024 Jireh Loreaux. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jireh Loreaux
+-/
+import Mathlib.LinearAlgebra.Dual
+import Mathlib.Analysis.NormedSpace.HahnBanach.Separation
+import Mathlib.Topology.Algebra.Module.WeakDual
+
+/-! # Closures of convex sets in locally convex spaces
+
+This file contains the standard result that if `E` is a vector space with two locally convex
+topologies, then the closure of a convex set is the same in either topology, provided they have the
+same collection of continuous linear functionals. In particular, the weak closure of a convex set
+in locally convex space coincides with the closure in the original topology.
+
+Of course, we phrase this in terms of linear maps between locally convex spaces, rather than
+creating two separate topologies on the same space.
+-/
+
+variable {𝕜 E F : Type*}
+variable [RCLike 𝕜] [AddCommGroup E] [Module 𝕜 E] [AddCommGroup F] [Module 𝕜 F]
+variable [Module ℝ E] [IsScalarTower ℝ 𝕜 E] [Module ℝ F] [IsScalarTower ℝ 𝕜 F]
+variable [TopologicalSpace E] [TopologicalAddGroup E] [ContinuousSMul 𝕜 E] [LocallyConvexSpace ℝ E]
+variable [TopologicalSpace F] [TopologicalAddGroup F] [ContinuousSMul 𝕜 F] [LocallyConvexSpace ℝ F]
+
+variable (𝕜) in
+/-- If `E` is a locally convex space over `𝕜` (with `RCLike 𝕜`), and `s : Set E` is `ℝ`-convex, then
+the closure of `s` and the weak closure of `s` coincide. More precisely, the topological closure
+commutes with `toWeakSpace 𝕜 E`.
+
+This holds more generally for any linear equivalence `e : E ≃ₗ[𝕜] F` between locally convex spaces
+such that precomposition with `e` and `e.symm` preserves continuity of linear functionals. See
+`LinearEquiv.image_closure_of_convex`. -/
```

Comment:
Perhaps instead:

See`LinearEquiv.image_closure_of_convex` below.

Although this assumes that the file will not be reorganized, it isn't obvious to the reader that the result in question will lie in this file and so it isn't obvious to look for it in what follows.

Reply:
That shouldn't matter much. It will be clickable in the docs, and in the editor you could just grep.